waspc/src/Wasp/Generator/JsImport.hs

```diff
@@ -43,9 +46,16 @@ jsImportToImportJson maybeJsImport = maybe notDefinedValue mkTmplData maybeJsImp
 
     mkTmplData :: JsImport -> Aeson.Value
     mkTmplData jsImport =
-      let (jsImportStmt, jsImportIdentifier) = getJsImportStmtAndIdentifier jsImport
+      let (jsImportStmt, jsImportIdentifier) = getJsImportStmtAndIdentifier $ mangleImportIdentifier jsImport
        in object
             [ "isDefined" .= True,
               "importStatement" .= jsImportStmt,
               "importIdentifier" .= jsImportIdentifier
             ]
+      where
+        mangleImportIdentifier :: JsImport -> JsImport
+        mangleImportIdentifier JI.JsImport {JI._name = JsImportModule name} = prefixTheImportName name jsImport
+        mangleImportIdentifier JI.JsImport {JI._name = JsImportField name} = prefixTheImportName name jsImport
+
+        prefixTheImportName :: String -> JsImport -> JsImport
+        prefixTheImportName originalName = applyJsImportAlias (Just ("__userDefined" ++ toUpperFirst originalName))
```

Comment:
I wanted to suggest mangling our imports instead of user imports (to keep the path between user symbols and our code as clear as possible).

However,  it seems that it's not our imports conflicting with user imports - it's the thing we're defining in the same file.

Anyway, I don't how to make this without changing user symbol names, but am leaving this comment in case you have an idea.